### PR TITLE
[Multi-CassKop] GKE Sample

### PR DESCRIPTION
**Pull request by erdrix**
Add a tutorial to setup Multi-Casskop on GKE (use terraform and helm).

### CODE REVIEW

File: multi-casskop/docs/tutorials/gke.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+$ export NAMESPACE=cassandra-demo
+$ export DNS_ZONE_NAME=external-dns-test-gcp-trycatchlearn-fr     # -> change with your own one
+$ export DNS_NAME=external-dns-test.gcp.trycatchlearn.fr          # -> change with your own one
+$ export MANAGED_ZONE=tracking-pdb                                # -> change with your own one
+```
+
+### Setup base infrastructure 
+
+```sh
+$ cd ${WORKSPACE}/multi-casskop/samples/gke/terraform
```
Comment (cscetbon) [inline multi-casskop/docs/tutorials/gke.md]: this variable could be confusing with the terraform workspace.

File: multi-casskop/docs/tutorials/gke.md [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+$ gcloud dns record-sets transaction start --zone "tracking-pdb"
+$ gcloud dns record-sets transaction add ns-cloud-d{1..4}.googledomains.com. \
+    --name "external-dns-test.gcp.trycatchlearn.fr." --ttl 300 --type NS --zone "tracking-pdb"
+$ gcloud dns record-sets transaction execute --zone "tracking-pdb"
+```
+
+To test external dns works correctly : 
+
+```sh
+cat <<EOF | sed "s/\[NAMESPACE\]/$NAMESPACE/" | kubectl apply -f -
```
Comment (cscetbon) [inline multi-casskop/docs/tutorials/gke.md]: There is no `[NAMESPACE]` in the following yaml ...

File: multi-casskop/samples/gke/terraform/cloud-dns.tf [hunk 3]
```diff
@@ -0,0 +1,134 @@
+// Documentations support : https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/gke.md
```
Comment (cscetbon) [inline multi-casskop/samples/gke/terraform/cloud-dns.tf]: Why not just use the Kubernetes deployment files we have to create those accounts ?
We could keep Terraform for the infrastructure but not for Kubernetes content ?

File: multi-casskop/samples/gke/terraform/gke-cluster.tf [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    # Whether to enable Cloud TPU resources in this cluster
+    enable_tpu = null
+
+    # The logging service that the cluster should write logs to. Available options include logging.googleapis.com, logging.googleapis.com/kubernetes, and none
+    logging_service = "logging.googleapis.com/kubernetes"
+
+    # The monitoring service that the cluster should write metrics to
+    monitoring_service = "monitoring.googleapis.com/kubernetes"
+    
+    # The maintenance policy to use for the cluster
```
Comment (cscetbon) [inline multi-casskop/samples/gke/terraform/gke-cluster.tf]: Let's remove default settings that are commented out

File: multi-casskop/samples/gke/terraform/kubernetes.tf [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+resource "helm_release" "casskop" {
+  name             = "casskop"
+  repository       = data.helm_repository.casskop.metadata[0].name
+  chart            = "cassandra-operator"
+  namespace        = kubernetes_namespace.cassandra-demo.metadata[0].name
+  disable_webhooks = false
+  #version          = "0.5.0-release"
+  set {
+    name  = "image.tag"
+    value = "v0.5.0-release"
```
Comment (cscetbon) [inline multi-casskop/samples/gke/terraform/kubernetes.tf]: Let's parameterize it. See https://www.terraform.io/docs/configuration/variables.html#declaring-an-input-variable

File: multi-casskop/samples/gke/terraform/cloud-dns.tf (hunk 3 above)
Comment (erdrix) [reply multi-casskop/samples/gke/terraform/cloud-dns.tf]: The aim of this tutorial is to focus on the Multi-CassKop deployment and usage, I don't want to add complexity and risks of mistakes. With terraform provisioning also the casskop and external dns, I ensure that the requirements to instantiate Multi-casskop will be fullfilled

Comment (cscetbon) [reply multi-casskop/samples/gke/terraform/cloud-dns.tf]: Yep except not everybody uses terraform .. up to you

Comment (fdehay) [reply multi-casskop/samples/gke/terraform/cloud-dns.tf]: In my humble opinion, I think Alexandre adds value in the form of a terraform deployment of casskop. Also he can focus on MCK testing. I would vote for keeping it as it is

Comment (cscetbon) [reply multi-casskop/samples/gke/terraform/cloud-dns.tf]: I’m not proposing to drop it but to keep kubernetes changes at the object level
In yaml files. In short if people don’t use terraform they can’t use this code 🤷‍♂️

Comment (fdehay) [reply multi-casskop/samples/gke/terraform/cloud-dns.tf]: Yep I know I was just thinking a complete terraform example might be interesting too. terraform is widely spread I like to believe, at least it is here. Anyway, I let you decide but thanks for your detailed reveiw on this one again!

**Pull request merged**